storage-access.bs

```diff
@@ -98,6 +98,18 @@ Script in the <{iframe}> can call |doc|`.`{{Document/hasStorageAccess()}} to det
 
 </div>
 
+This specification also defines a method for a [=site=] to request access to its [=unpartitioned data=] when later loaded in a [=third party context=] on another specified [=site=] ({{Document/requestStorageAccessUnderSite()}}), and a method for a [=site=] to allow such requests to succeed ({{Document/allowStorageAccessRequestOnSite()}}).
+
+<div class=example>
+
+Alex visits `https://site.example`. Alex clicks "Login with Identity", which Alex expects to log them in with their account on `https://identity.example`. The page calls `{{Document.allowStorageAccessRequestOnSite("https://identity.example")}}` and opens a new [=window=] for `https://identity.example`.
+
+In the new [=window=], Alex selects the account to log in with. In response, `https://identity.example` sets a cookie, which by virtue of its [=first-party-site context=] is in its [=unpartitioned data=] and calls `{{Document.requestStorageAccessUnderSite("https://site.example")}}`. Once the {{Promise}} returned by `{{Document/requestStorageAccessUnderSite()}}` resolves, the `https://identity.example` [=window=] closes itself, returning Alex to `https://site.example`.
+
+When Alex returns to `https://site.example`, the {{Promise}} returned by `{{Document/allowStorageAccessRequestOnSite()}}` has resolved or will resolve shortly. At this point, requests to `https://idnetity.example` will bear the cookie set in the previous [=window=].
```

Comment:
To be clear, is this saying `allowStorageAccessRequestOnSite` would remaining pending until `requestStorageAccessUnderSite` is invoked? Note that while authentication flows can happen in popups, they could be done via a full-page redirect.

Reply:
That was the intent. Given that, I think it would be best to have that Promise resolve once the internal state is set to allow the `requestStorageAccessUnderSite` call to succeed.

Reply:
Right, I don't fully understand why the Promise needs to be left dangling on site.example. What was the reason for that?

Reply:
I wanted a means to communicate the timing of unpartitioned storage availability in the `https://site.example` page. How can an iframe that is waiting for a separate window to grant it unpartitioned storage know when it has that available? I think the answer right now is polling. Perhaps we want a new function, e.g. `waitForStorageAccess()` ?

Regardless, I think that this promise should resolve once the intemediate state of "ready for a  `requestStorageAccessUnderSite` call" is set.

Reply:
Yup, this is #55 and I agree it gets even more interesting with the popup use case.